### ISSUE TITLE
Lecture 10

### ISSUE THREAD

**Issue by yoavram** [yoavram/Py4Life#16]
Finish it

**Comment by yoavram** [comment]
Lecture 10 is ready, @soungalo can you review it please?

**Comment by yoavram** [comment]
Also I'm thinking of changing between 8 and 10, what do you think, @soungalo ?

**Comment by yoavram** [comment]
Add reference: DataFrame / tidy data - http://www.jstatsoft.org/v59/i10

**Comment by soungalo** [comment]
Very interesting! Just a few notes:
1. When doing the _melt_ on the dataframe, I think it needs more explanation. It helped me to run df.shape.
2. When showing the equations of the exponential model, explain the parameters _N_ and _r_.
3. The videos don't work.
4. "Here we do a hack to sort the countires names based on the largest population in 2013:" - why do you do that? and I think the code needs more explanation.
5. After showing the fit of the linear model to the log population and the logistic model to the bacteria growth, maybe show some measurement of the deviation from the model, because it's not very clear just by looking at the plot.
6. When solving the ODE, I think you should show the general idea of how ODEs are solved. You don't have to do the actual integration, but just show that you are integrating. Keep in mind that most students have never seen an ODE before.
7. When loading the 96-well data, just go for the url. No one will have the file on their local system.
8. I think we've never showed them list comprehensions before, so either skip it and tell them to look into it if they want or spend some time explaining the syntax.

**Comment by yoavram** [comment]
Thanks!
1. done
2. done
3. strange, they sometime work. I'll check it.
4. done
5. printing correlation coefficient and p-value, added ref to calculate R^2
6. that's what the youtube does, I'm not going to explain numerical integration. 
7. I do the `os.path.exists` so that if I already copied the file (like if I already ran the notebook) it won't load it again. Seems reasonable, and shows them about `os.path.exists`.
8. I'll see if there is time.
